<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evgenii Tretiakov" />

<meta name="date" content="2023-05-29" />

<title>Exploratory analysis of Cnr1 in mouse Onecut3-GABAergic and dopaminergic hypothalamic neurons during development</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PeVN-dopaminergic-Cnr1</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="eda.html">Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="90-methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/Harkany-Lab/PeVN-dopaminergic-Cnr1">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Exploratory analysis of Cnr1 in mouse
Onecut3-GABAergic and dopaminergic hypothalamic neurons during
development</h1>
<h4 class="author">Evgenii Tretiakov</h4>
<h4 class="date">2023-05-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PeVN-dopaminergic-Cnr1/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has staged changes. To know which version of the
R Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230529code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230529)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230529code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230529)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHarkanyLabPeVNdopaminergicCnr1tree66a65aed0f719734002393c8a81d391d1e779179targetblank66a65aea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Harkany-Lab/PeVN-dopaminergic-Cnr1/tree/66a65aed0f719734002393c8a81d391d1e779179" target="_blank">66a65ae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomHarkanyLabPeVNdopaminergicCnr1tree66a65aed0f719734002393c8a81d391d1e779179targetblank66a65aea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Harkany-Lab/PeVN-dopaminergic-Cnr1/tree/66a65aed0f719734002393c8a81d391d1e779179" target="_blank">66a65ae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/
    Ignored:    output/tables/

Staged changes:
    Modified:   analysis/_site.yml
    New:        analysis/eda.Rmd
    New:        code/functions.R
    New:        code/genes.R
    New:        data/.gitattributes
    New:        data/colours_wtree.tsv
    New:        data/oc3_fin.h5seurat
    New:        data/oldCCA_nae_srt.rds
    New:        data/rar2020.srt.cont.oc2or3.raw.h5seurat
    New:        data/samples.tsv
    New:        output/figures/stat-corr-plt_oc3-rna-data-Onecut3-Cnr1_.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code># Load tidyverse infrastructure packages
library(here)
library(tidyverse)
library(magrittr)
library(zeallot)
library(future)

# Load packages for scRNA-seq analysis and visualisation
library(sctransform)
library(Seurat)
library(SeuratWrappers)
library(SeuratDisk)
library(scCustomize)
library(UpSetR)
library(patchwork)
library(RColorBrewer)
library(Nebulosa)</code></pre>
<pre class="r"><code>src_dir &lt;- here(&quot;code&quot;)
data_dir &lt;- here(&quot;data&quot;)
output_dir &lt;- here(&quot;output&quot;)
plots_dir &lt;- here(output_dir, &quot;figures&quot;)
tables_dir &lt;- here(output_dir, &quot;tables&quot;)</code></pre>
<pre class="r"><code>source(here(src_dir, &quot;functions.R&quot;))
source(here(src_dir, &quot;genes.R&quot;))</code></pre>
<pre class="r"><code>reseed &lt;- 42
set.seed(seed = reseed)</code></pre>
<pre class="r"><code># available cores
n_cores &lt;- available_cores(prop2use = .5)
# Parameters for parallel execution
plan(&quot;multicore&quot;, workers = n_cores)
options(
  future.globals.maxSize = Inf,
  future.rng.onMisuse = &quot;ignore&quot;
)
plan()</code></pre>
<pre><code>multicore:
- args: function (..., workers = 16, envir = parent.frame())
- tweaked: TRUE
- call: plan(&quot;multicore&quot;, workers = n_cores)</code></pre>
<div id="read-data" class="section level2">
<h2>Read data</h2>
<pre class="r"><code>rar2020_ages_all &lt;- c(&quot;E15&quot;, &quot;E17&quot;, &quot;P00&quot;, &quot;P02&quot;, &quot;P10&quot;, &quot;P23&quot;)
rar2020_ages_postnat &lt;- c(&quot;P02&quot;, &quot;P10&quot;, &quot;P23&quot;)
samples_df &lt;- read_tsv(here(&quot;data/samples.tsv&quot;))
colours_wtree &lt;- setNames(
  read_lines(here(data_dir, &quot;colours_wtree.tsv&quot;)),
  1:45
)

rar2020_srt_pub &lt;-
  readr::read_rds(file.path(data_dir, &quot;oldCCA_nae_srt.rds&quot;))
rar2020_srt_pub %&lt;&gt;% UpdateSeuratObject()
colnames(rar2020_srt_pub@reductions$umap@cell.embeddings) &lt;-
  c(&quot;UMAP_1&quot;, &quot;UMAP_2&quot;)

rar2020_srt_pub$orig.ident &lt;-
  rar2020_srt_pub %&gt;%
  colnames() %&gt;%
  str_split(pattern = &quot;:&quot;, simplify = TRUE) %&gt;%
  .[, 1] %&gt;%
  plyr::mapvalues(
    x = .,
    from = samples_df$fullname,
    to = samples_df$sample
  )
rar2020_srt_pub$age &lt;-
  plyr::mapvalues(
    x = rar2020_srt_pub$orig.ident,
    from = samples_df$sample,
    to = samples_df$age
  )
Idents(rar2020_srt_pub) &lt;- &quot;wtree&quot;
neurons &lt;-
  subset(rar2020_srt_pub, idents = c(&quot;18&quot;, &quot;32&quot;))
neurons &lt;-
  subset(neurons, subset = Slc17a6 == 0)
onecut3 &lt;-
  subset(neurons,
    subset = Onecut3 &gt; 0 | Th &gt; 0 | Ddc &gt; 0 | Slc6a3 &gt; 0
  )

onecut3 &lt;-
  Store_Palette_Seurat(
    seurat_object = onecut3,
    palette = rev(brewer.pal(n = 11, name = &quot;Spectral&quot;)),
    palette_name = &quot;div_Colour_Pal&quot;
  )</code></pre>
<p>Thus we subset the dataset to the neurons of interest from cluster
18, which are GABAergic and dopaminergic neurons expressing Onecut3
transcription factor. We also explicitly exclude Slc17a6-expressing
neurons, which are glutamatergic neurons just in case to reduce noise.
As the control group we use dopaminergic TIDA-neurons from the arcuate
nucleus.</p>
</div>
<div id="derive-and-filter-matrix-of-neurons-of-interest"
class="section level2">
<h2>Derive and filter matrix of neurons of interest</h2>
<pre class="r"><code>mtx_oc3 &lt;-
  onecut3 %&gt;%
  GetAssayData(&quot;data&quot;, &quot;RNA&quot;) %&gt;%
  as.data.frame() %&gt;%
  t()
rownames(mtx_oc3) &lt;- colnames(onecut3)

# Filter features
filt_low_genes &lt;-
  colSums(mtx_oc3) %&gt;%
  .[. &gt; quantile(., 0.4)] %&gt;%
  names()
mtx_oc3 %&lt;&gt;% .[, filt_low_genes]

min_filt_vector &lt;-
  mtx_oc3 %&gt;%
  as_tibble() %&gt;%
  select(all_of(filt_low_genes)) %&gt;%
  summarise(across(.fns = ~ quantile(.x, .1))) %&gt;%
  as.list() %&gt;%
  map(as.double) %&gt;%
  simplify() %&gt;%
  .[colnames(mtx_oc3)]

# Prepare table of intersection sets analysis
content_mtx_oc3 &lt;-
  (mtx_oc3 &gt; min_filt_vector) %&gt;%
  as_tibble() %&gt;%
  mutate_all(as.numeric)</code></pre>
</div>
<div id="plot-umaps-density" class="section level2">
<h2>Plot UMAPs density</h2>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Onecut3&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-onecut3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Th&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Th-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Slc6a3&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Slc6a3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Prlr&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Prlr-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Cnr1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Cnr1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Gpr55&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Gpr55-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Mgll&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Mgll-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Dagla&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Dagla-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Daglb&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Daglb-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Faah&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Faah-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Napepld&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Napepld-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Gde1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Gde1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Pparg&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Pparg-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Slc32a1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-slc32a1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Gad1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-gad1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Custom(
  seurat_object = onecut3,
  features = c(&quot;Gad2&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-gad2-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="joint-density-umaps-plots" class="section level2">
<h2>Joint density UMAP’s plots</h2>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Onecut3&quot;, &quot;Cnr1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Onecut3-Cnr1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Onecut3&quot;, &quot;Th&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Onecut3-Th-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Cnr1&quot;, &quot;Th&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Cnr1-Th-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Prlr&quot;, &quot;Cnr1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Prlr-Cnr1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Prlr&quot;, &quot;Th&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Prlr-Th-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Prlr&quot;, &quot;Onecut3&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Prlr-Onecut3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Gpr55&quot;, &quot;Prlr&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Gpr55-Prlr-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Gpr55&quot;, &quot;Cnr1&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Gpr55-Cnr1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Gpr55&quot;, &quot;Th&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Gpr55-Th-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Plot_Density_Joint_Only(
  seurat_object = onecut3,
  features = c(&quot;Gpr55&quot;, &quot;Onecut3&quot;),
  custom_palette = onecut3@misc$div_Colour_Pal
)</code></pre>
<p><img src="figure/eda.Rmd/plt-init-umap-density-Gpr55-Onecut3-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation-analysis-visualisation-between-different-genes"
class="section level2">
<h2>Correlation analysis visualisation between different genes</h2>
<pre class="r"><code>p_corrs &lt;- list(
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    x = Onecut3,
    y = Cnr1,
    xfill = &quot;#ffc400&quot;,
    yfill = &quot;#e22ee2&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    x = Slc32a1,
    y = Onecut3,
    xfill = &quot;#0000da&quot;,
    yfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    x = Gpr55,
    y = Onecut3,
    xfill = &quot;#006eff&quot;,
    yfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    x = Slc32a1,
    y = Cnr1,
    xfill = &quot;#0000da&quot;,
    yfill = &quot;#e22ee2&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    x = Gpr55,
    y = Cnr1,
    xfill = &quot;#006eff&quot;,
    yfill = &quot;#e22ee2&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    x = Slc32a1,
    y = Gpr55,
    xfill = &quot;#0000da&quot;,
    yfill = &quot;#006eff&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Slc32a1,
    x = Onecut3,
    yfill = &quot;#0000da&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Gpr55,
    x = Onecut3,
    yfill = &quot;#006eff&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Th,
    x = Onecut3,
    yfill = &quot;#ff0000&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Gad1,
    x = Onecut3,
    yfill = &quot;#a50202&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Gad2,
    x = Onecut3,
    yfill = &quot;#4002a5&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Onecut2,
    x = Onecut3,
    yfill = &quot;#6402a5&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Prlr,
    x = Onecut3,
    yfill = &quot;#2502a5&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Ddc,
    x = Onecut3,
    yfill = &quot;#4002a5&quot;,
    xfill = &quot;#ffc400&quot;
  ),
  ggstatsplot::ggscatterstats(
    as.data.frame(mtx_oc3),
    y = Slc6a3,
    x = Onecut3,
    yfill = &quot;#2502a5&quot;,
    xfill = &quot;#ffc400&quot;
  )
)
n_corrs &lt;- list(
  &quot;oc3-rna-data-Onecut3-Cnr1&quot;,
  &quot;oc3-rna-data-Slc32a1-Onecut3&quot;,
  &quot;oc3-rna-data-Gpr55-Onecut3&quot;,
  &quot;oc3-rna-data-Slc32a1-Cnr1&quot;,
  &quot;oc3-rna-data-Gpr55-Cnr1&quot;,
  &quot;oc3-rna-data-Slc32a1-Gpr55&quot;,
  &quot;oc3-rna-data-Onecut3-Slc32a1&quot;,
  &quot;oc3-rna-data-Onecut3-Gpr55&quot;,
  &quot;oc3-rna-data-Onecut3-Th&quot;,
  &quot;oc3-rna-data-Onecut3-Gad1&quot;,
  &quot;oc3-rna-data-Onecut3-Gad2&quot;,
  &quot;oc3-rna-data-Onecut3-Onecut2&quot;,
  &quot;oc3-rna-data-Onecut3-Prlr&quot;,
  &quot;oc3-rna-data-Onecut3-Ddc&quot;,
  &quot;oc3-rna-data-Onecut3-Slc6a3&quot;
)

walk2(n_corrs, p_corrs, save_my_plot, type = &quot;stat-corr-plt&quot;)</code></pre>
<div
id="visualise-intersections-sets-that-we-are-going-to-use-highlighted"
class="section level3">
<h3>Visualise intersections sets that we are going to use
(highlighted)</h3>
<pre class="r"><code>upset(
  as.data.frame(content_mtx_oc3),
  order.by = &quot;freq&quot;,
  sets.x.label = &quot;Number of cells&quot;,
  text.scale = c(2, 1.6, 2, 1.3, 2, 3),
  nsets = 15,
  sets = c(&quot;Th&quot;, &quot;Onecut3&quot;),
  queries = list(
    list(
      query = intersects,
      params = list(&quot;Onecut3&quot;),
      active = T
    ),
    list(
      query = intersects,
      params = list(&quot;Th&quot;),
      active = T
    ),
    list(
      query = intersects,
      params = list(&quot;Th&quot;, &quot;Onecut3&quot;),
      active = T
    )
  ),
  nintersects = 60,
  empty.intersections = &quot;on&quot;
)</code></pre>
<p><img src="figure/eda.Rmd/plot-upset-oc3-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(
  as.data.frame(content_mtx_oc3),
  order.by = &quot;freq&quot;,
  sets.x.label = &quot;Number of cells&quot;,
  text.scale = c(2, 1.6, 2, 1.3, 2, 3),
  nsets = 15,
  sets = c(&quot;Gpr55&quot;, &quot;Cnr1&quot;, &quot;Slc32a1&quot;, &quot;Th&quot;),
  queries = list(
    list(
      query = intersects,
      params = list(&quot;Gpr55&quot;, &quot;Slc32a1&quot;),
      active = T
    ),
    list(
      query = intersects,
      params = list(&quot;Cnr1&quot;, &quot;Th&quot;),
      active = T
    )
  ),
  nintersects = 60,
  empty.intersections = &quot;on&quot;
)</code></pre>
<p><img src="figure/eda.Rmd/plot-upset-oc3-mod-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>Regroup factor by stages for more balanced groups</p>
<pre class="r"><code>onecut3$age %&gt;% forcats::fct_count()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["f"],"name":[1],"type":["fct"],"align":["left"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"E15","2":"85"},{"1":"E17","2":"71"},{"1":"P00","2":"48"},{"1":"P02","2":"44"},{"1":"P10","2":"145"},{"1":"P23","2":"8"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>onecut3$stage &lt;-
  onecut3$age %&gt;%
  forcats::fct_collapse(
    `Embrionic day 15` = &quot;E15&quot;,
    `Embrionic day 17` = &quot;E17&quot;,
    Neonatal = c(&quot;P00&quot;, &quot;P02&quot;),
    Postnatal = c(&quot;P10&quot;, &quot;P23&quot;)
  )
onecut3$stage %&gt;% forcats::fct_count()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["f"],"name":[1],"type":["fct"],"align":["left"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"Embrionic day 15","2":"85"},{"1":"Embrionic day 17","2":"71"},{"1":"Neonatal","2":"92"},{"1":"Postnatal","2":"153"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Make subset of stable neurons</p>
<pre class="r"><code>onecut3$gaba_status &lt;-
  content_mtx_oc3 %&gt;%
  select(Gad1, Gad2, Slc32a1) %&gt;%
  mutate(gaba = if_all(.fns = ~ .x &gt; 0)) %&gt;%
  .$gaba

onecut3$gaba_occurs &lt;-
  content_mtx_oc3 %&gt;%
  select(Gad1, Gad2, Slc32a1) %&gt;%
  mutate(gaba = if_any(.fns = ~ .x &gt; 0)) %&gt;%
  .$gaba

onecut3$th_status &lt;-
  content_mtx_oc3 %&gt;%
  select(Th, Ddc, Slc6a3) %&gt;%
  mutate(dopamin = if_any(.fns = ~ .x &gt; 0)) %&gt;%
  .$dopamin

oc3_fin &lt;- onecut3</code></pre>
</div>
</div>
<div id="check-contingency-tables-for-neurotransmitter-signature"
class="section level2">
<h2>Check contingency tables for neurotransmitter signature</h2>
<pre class="r"><code>oc3_fin@meta.data %&gt;%
  janitor::tabyl(th_status, gaba_status)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["th_status"],"name":[1],"type":["lgl"],"align":["right"]},{"label":["FALSE"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["TRUE"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"FALSE","2":"136","3":"58"},{"1":"TRUE","2":"150","3":"57"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>By age</p>
<pre class="r"><code>oc3_fin@meta.data %&gt;%
  janitor::tabyl(age, th_status)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["age"],"name":[1],"type":["chr"],"align":["left"]},{"label":["FALSE"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["TRUE"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"E15","2":"46","3":"39"},{"1":"E17","2":"36","3":"35"},{"1":"P00","2":"37","3":"11"},{"1":"P02","2":"14","3":"30"},{"1":"P10","2":"59","3":"86"},{"1":"P23","2":"2","3":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>By stage</p>
<pre class="r"><code>oc3_fin@meta.data %&gt;%
  janitor::tabyl(stage, th_status)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["FALSE"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["TRUE"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"Embrionic day 15","2":"46","3":"39"},{"1":"Embrionic day 17","2":"36","3":"35"},{"1":"Neonatal","2":"51","3":"41"},{"1":"Postnatal","2":"61","3":"92"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Make splits of neurons by neurotransmitter signature</p>
<pre class="r"><code>oc3_fin$status &lt;- oc3_fin$th_status %&gt;%
  if_else(true = &quot;dopaminergic&quot;,
    false = &quot;GABAergic&quot;
  )
Idents(oc3_fin) &lt;- &quot;status&quot;
SaveH5Seurat(
  object    = oc3_fin,
  filename  = here(data_dir, &quot;oc3_fin&quot;),
  overwrite = TRUE,
  verbose   = TRUE
)

## Split on basis of neurotrans and test for difference
oc3_fin_neurotrans &lt;- SplitObject(oc3_fin, split.by = &quot;status&quot;)

## Split on basis of age and test for difference
oc3_fin_ages &lt;- SplitObject(oc3_fin, split.by = &quot;age&quot;)</code></pre>
</div>
<div id="dotplots-grouped-by-age" class="section level2">
<h2>DotPlots grouped by age</h2>
<div
id="expression-of-gaba-receptors-in-gabaergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of GABA receptors in GABAergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$GABAergic,
  features = gabar,
  group.by = &quot;age&quot;,
  cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-age-gabar-gaba-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-gaba-receptors-in-dopaminergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of GABA receptors in dopaminergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$dopaminergic,
  features = gabar,
  group.by = &quot;age&quot;,
  cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-age-gabar-dop-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-glutamate-receptors-in-gabaergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of glutamate receptors in GABAergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$GABAergic,
  features = glutr,
  group.by = &quot;age&quot;,
  cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-age-glur-gaba-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-glutamate-receptors-in-dopaminergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of glutamate receptors in dopaminergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$dopaminergic,
  features = glutr,
  group.by = &quot;age&quot;,
  cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-age-glur-dop-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dotplots-grouped-by-stage" class="section level2">
<h2>DotPlots grouped by stage</h2>
<div
id="expression-of-gaba-receptors-in-gabaergic-onecut3-positive-cells-1"
class="section level3">
<h3>Expression of GABA receptors in GABAergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$GABAergic,
  features = gabar,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-gabar-gaba-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-gaba-receptors-in-dopaminergic-onecut3-positive-cells-1"
class="section level3">
<h3>Expression of GABA receptors in dopaminergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$dopaminergic,
  features = gabar,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-gabar-dop-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-glutamate-receptors-in-gabaergic-onecut3-positive-cells-1"
class="section level3">
<h3>Expression of glutamate receptors in GABAergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$GABAergic,
  features = glutr,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-glur-gaba-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-glutamate-receptors-in-dopaminergic-onecut3-positive-cells-1"
class="section level3">
<h3>Expression of glutamate receptors in dopaminergic Onecut3 positive
cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$dopaminergic,
  features = glutr,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-glur-dop-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-neuromodulators-receptors-in-gabaergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of neuromodulators receptors in GABAergic Onecut3
positive cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$GABAergic,
  features = npr,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-npr-gaba-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-neuromodulators-receptors-in-dopaminergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of neuromodulators receptors in dopaminergic Onecut3
positive cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$dopaminergic,
  features = npr,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-npr-dop-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-endocannabinoids-relevant-genes-in-gabaergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of endocannabinoids relevant genes in GABAergic Onecut3
positive cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$GABAergic,
  features = cnbn,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-cnbn-gaba-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="expression-of-endocannabinoids-relevant-genes-in-dopaminergic-onecut3-positive-cells"
class="section level3">
<h3>Expression of endocannabinoids relevant genes in dopaminergic
Onecut3 positive cells</h3>
<pre class="r"><code>DotPlot(
  object = oc3_fin_neurotrans$dopaminergic,
  features = cnbn,
  group.by = &quot;stage&quot;,
  cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
  col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/eda.Rmd/plot-dotplot-stage-cnbn-dop-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="overrepresentation-analysis" class="section level2">
<h2>Overrepresentation analysis</h2>
<pre class="r"><code>Idents(oc3_fin) &lt;- &quot;status&quot;

sbs_mtx_oc &lt;-
  oc3_fin %&gt;%
  GetAssayData(&quot;data&quot;, &quot;RNA&quot;) %&gt;%
  as.data.frame() %&gt;%
  t()
rownames(sbs_mtx_oc) &lt;- colnames(oc3_fin)

# Filter features
filt_low_genes2 &lt;-
  colSums(sbs_mtx_oc) %&gt;%
  .[. &gt; quantile(., 0.4)] %&gt;%
  names()
sbs_mtx_oc %&lt;&gt;% .[, filt_low_genes2]

min_filt_vector2 &lt;-
  sbs_mtx_oc %&gt;%
  as_tibble() %&gt;%
  select(all_of(filt_low_genes2)) %&gt;%
  summarise(across(.fns = ~ quantile(.x, .005))) %&gt;%
  as.list() %&gt;%
  map(as.double) %&gt;%
  simplify() %&gt;%
  .[filt_low_genes2]

# Prepare table of intersection sets analysis
content_sbs_mtx_oc &lt;-
  (sbs_mtx_oc &gt; min_filt_vector2) %&gt;%
  as_tibble() %&gt;%
  mutate_all(as.numeric)</code></pre>
<pre class="r"><code>upset(
  as.data.frame(content_sbs_mtx_oc),
  order.by = &quot;freq&quot;,
  sets.x.label = &quot;Number of cells&quot;,
  text.scale = c(2, 1.6, 2, 1.3, 2, 3),
  nsets = 15,
  sets = c(
    &quot;Gad1&quot;, &quot;Gad2&quot;, &quot;Slc32a1&quot;,
    &quot;Th&quot;, &quot;Onecut3&quot;,
    cnbn, &quot;Prlr&quot;
  ) %&gt;%
    .[. %in% colnames(content_sbs_mtx_oc)],
  nintersects = 20,
  empty.intersections = NULL
)</code></pre>
<p><img src="figure/eda.Rmd/plot-upset-sbs-oc3-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(
  as.data.frame(content_sbs_mtx_oc),
  order.by = &quot;freq&quot;,
  sets.x.label = &quot;Number of cells&quot;,
  text.scale = c(2, 1.6, 2, 1.3, 2, 3),
  nsets = 15,
  sets = c(cnbn, &quot;Prlr&quot;, &quot;Slc32a1&quot;, &quot;Th&quot;, &quot;Onecut3&quot;) %&gt;%
    .[. %in% colnames(content_sbs_mtx_oc)],
  nintersects = 10,
  empty.intersections = NULL
)</code></pre>
<p><img src="figure/eda.Rmd/plot-upset-sbs-oc3-mod-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sbs_mtx_oc_full &lt;- content_sbs_mtx_oc |&gt;
  select(any_of(c(
    neurotrans, cnbn, &quot;Prlr&quot;, &quot;Cnr1&quot;, &quot;Gpr55&quot;, &quot;Onecut3&quot;
  ))) |&gt;
  dplyr::bind_cols(oc3_fin@meta.data)

sbs_mtx_oc_full |&gt; glimpse()</code></pre>
<pre><code>Rows: 401
Columns: 43
$ Slc17a8          &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Slc1a1           &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,…
$ Slc1a2           &lt;dbl&gt; 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0,…
$ Slc1a6           &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Gad1             &lt;dbl&gt; 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1,…
$ Slc32a1          &lt;dbl&gt; 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,…
$ Slc6a1           &lt;dbl&gt; 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0,…
$ Cnr1             &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,…
$ Gpr55            &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Dagla            &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,…
$ Daglb            &lt;dbl&gt; 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Mgll             &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Faah             &lt;dbl&gt; 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1,…
$ Napepld          &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Gde1             &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1,…
$ Pparg            &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ Prlr             &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…
$ Onecut3          &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ nGene            &lt;int&gt; 3034, 2551, 2029, 1415, 1513, 1079, 3160, 3072, 2778,…
$ nUMI             &lt;dbl&gt; 6868, 5316, 3696, 2667, 2490, 1789, 8247, 8544, 6478,…
$ orig.ident       &lt;chr&gt; &quot;FC2P23&quot;, &quot;FC2P23&quot;, &quot;FC2P23&quot;, &quot;FC2P23&quot;, &quot;FC2P23&quot;, &quot;FC…
$ res.0.2          &lt;chr&gt; &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;, &quot;5&quot;…
$ res.0.4          &lt;chr&gt; &quot;26&quot;, &quot;26&quot;, &quot;26&quot;, &quot;26&quot;, &quot;26&quot;, &quot;26&quot;, &quot;26&quot;, &quot;26&quot;, &quot;26&quot;,…
$ res.0.8          &lt;chr&gt; &quot;31&quot;, &quot;31&quot;, &quot;31&quot;, &quot;31&quot;, &quot;31&quot;, &quot;31&quot;, &quot;31&quot;, &quot;31&quot;, &quot;31&quot;,…
$ res.1.2          &lt;chr&gt; &quot;36&quot;, &quot;36&quot;, &quot;36&quot;, &quot;36&quot;, &quot;36&quot;, &quot;36&quot;, &quot;36&quot;, &quot;36&quot;, &quot;36&quot;,…
$ res.2            &lt;chr&gt; &quot;37&quot;, &quot;37&quot;, &quot;37&quot;, &quot;37&quot;, &quot;37&quot;, &quot;37&quot;, &quot;37&quot;, &quot;37&quot;, &quot;37&quot;,…
$ tree.ident       &lt;int&gt; 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 2…
$ pro_Inter        &lt;chr&gt; &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;,…
$ pro_Enter        &lt;chr&gt; &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;, &quot;17&quot;,…
$ tree_final       &lt;fct&gt; 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 12, 1…
$ subtree          &lt;fct&gt; 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 1…
$ prim_walktrap    &lt;fct&gt; 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 2…
$ umi_per_gene     &lt;dbl&gt; 2.263678, 2.083889, 1.821587, 1.884806, 1.645737, 1.6…
$ log_umi_per_gene &lt;dbl&gt; 0.3548147, 0.3188745, 0.2604499, 0.2752666, 0.2163604…
$ nCount_RNA       &lt;dbl&gt; 6868, 5316, 3696, 2667, 2490, 1789, 8247, 8544, 6478,…
$ nFeature_RNA     &lt;int&gt; 3034, 2551, 2029, 1415, 1513, 1079, 3160, 3072, 2778,…
$ wtree            &lt;fct&gt; 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 3…
$ age              &lt;chr&gt; &quot;P23&quot;, &quot;P23&quot;, &quot;P23&quot;, &quot;P23&quot;, &quot;P23&quot;, &quot;P23&quot;, &quot;P10&quot;, &quot;P10…
$ stage            &lt;fct&gt; Postnatal, Postnatal, Postnatal, Postnatal, Postnatal…
$ gaba_status      &lt;lgl&gt; FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, FALSE…
$ gaba_occurs      &lt;lgl&gt; TRUE, TRUE, FALSE, FALSE, TRUE, TRUE, TRUE, TRUE, TRU…
$ th_status        &lt;lgl&gt; TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE,…
$ status           &lt;chr&gt; &quot;dopaminergic&quot;, &quot;dopaminergic&quot;, &quot;dopaminergic&quot;, &quot;dopa…</code></pre>
<pre class="r"><code>sbs_mtx_oc_full$modulator &lt;-
  sbs_mtx_oc_full %&gt;%
  select(Cnr1, Gpr55) %&gt;%
  mutate(modulator = if_any(.fns = ~ .x &gt; 0)) %&gt;%
  .$modulator

sbs_mtx_oc_full$oc3 &lt;-
  (sbs_mtx_oc_full$Onecut3 &gt; 0)

library(ggstatsplot)
# for reproducibility
set.seed(123)

# plot
grouped_ggpiestats(
  # arguments relevant for `ggpiestats()`
  data = sbs_mtx_oc_full,
  x = modulator,
  y = oc3,
  grouping.var = status,
  perc.k = 1,
  package = &quot;ggsci&quot;,
  palette = &quot;category10_d3&quot;,
  # arguments relevant for `combine_plots()`
  title.text = &quot;Neuromodulator specification of onecut-driven hypothalamic neuronal lineages by Onecut3 and main neurotransmitter expression&quot;,
  caption.text = &quot;Asterisks denote results from proportion tests; \n***: p &lt; 0.001, ns: non-significant&quot;,
  plotgrid.args = list(nrow = 2)
)</code></pre>
<p><img src="figure/eda.Rmd/plot-piecharts-sbs-oc3-chisqr-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggstatsplot_0.11.1.9000 Nebulosa_1.10.0         RColorBrewer_1.1-3     
 [4] patchwork_1.1.2.9000    UpSetR_1.4.0            scCustomize_1.1.1      
 [7] SeuratDisk_0.0.0.9020   SeuratWrappers_0.3.1    SeuratObject_4.1.3     
[10] Seurat_4.3.0            sctransform_0.3.5       future_1.32.0          
[13] zeallot_0.1.0           magrittr_2.0.3          lubridate_1.9.2        
[16] forcats_1.0.0           stringr_1.5.0           dplyr_1.1.2            
[19] purrr_1.0.1             readr_2.1.4             tidyr_1.3.0            
[22] tibble_3.2.1            ggplot2_3.4.2           tidyverse_2.0.0.9000   
[25] here_1.0.1              workflowr_1.7.0        

loaded via a namespace (and not attached):
  [1] fs_1.6.2                    matrixStats_0.63.0         
  [3] spatstat.sparse_3.0-1       bitops_1.0-7               
  [5] httr_1.4.5                  insight_0.19.1             
  [7] tools_4.3.0                 utf8_1.2.3                 
  [9] R6_2.5.1                    statsExpressions_1.5.0     
 [11] mgcv_1.8-42                 lazyeval_0.2.2             
 [13] uwot_0.1.14                 withr_2.5.0                
 [15] sp_1.6-0                    gridExtra_2.3              
 [17] progressr_0.13.0            textshaping_0.3.6          
 [19] cli_3.6.1                   Biobase_2.60.0             
 [21] spatstat.explore_3.1-0      sandwich_3.0-2             
 [23] prismatic_1.1.1             labeling_0.4.2             
 [25] sass_0.4.5                  mvtnorm_1.1-3              
 [27] spatstat.data_3.0-1         ggridges_0.5.4             
 [29] pbapply_1.7-0               systemfonts_1.0.4          
 [31] R.utils_2.12.2              parallelly_1.35.0          
 [33] rstudioapi_0.14             generics_0.1.3             
 [35] shape_1.4.6                 ica_1.0-3                  
 [37] spatstat.random_3.1-4       vroom_1.6.1                
 [39] Matrix_1.5-4                ggbeeswarm_0.7.1.9000      
 [41] fansi_1.0.4                 S4Vectors_0.38.0           
 [43] abind_1.4-5                 R.methodsS3_1.8.2          
 [45] lifecycle_1.0.3             whisker_0.4.1              
 [47] multcomp_1.4-23             yaml_2.3.7                 
 [49] snakecase_0.11.0            SummarizedExperiment_1.30.0
 [51] Rtsne_0.16                  paletteer_1.5.0            
 [53] grid_4.3.0                  promises_1.2.0.1           
 [55] crayon_1.5.2                miniUI_0.1.1.1             
 [57] lattice_0.21-8              cowplot_1.1.1              
 [59] pillar_1.9.0                knitr_1.42                 
 [61] GenomicRanges_1.52.0        estimability_1.4.1         
 [63] future.apply_1.10.0         codetools_0.2-19           
 [65] leiden_0.4.3                glue_1.6.2                 
 [67] getPass_0.2-2               data.table_1.14.8          
 [69] remotes_2.4.2               vctrs_0.6.2                
 [71] png_0.1-8                   gtable_0.3.3               
 [73] rematch2_2.1.2              datawizard_0.7.1           
 [75] cachem_1.0.7                ks_1.14.0                  
 [77] xfun_0.39                   mime_0.12                  
 [79] correlation_0.8.4           ggside_0.2.2               
 [81] pracma_2.4.2                coda_0.19-4                
 [83] survival_3.5-5              SingleCellExperiment_1.22.0
 [85] ellipsis_0.3.2              fitdistrplus_1.1-11        
 [87] TH.data_1.1-2               ROCR_1.0-11                
 [89] nlme_3.1-162                bit64_4.0.5                
 [91] RcppAnnoy_0.0.20            GenomeInfoDb_1.36.0        
 [93] rprojroot_2.0.3             bslib_0.4.2                
 [95] irlba_2.3.5.1               vipor_0.4.5                
 [97] KernSmooth_2.23-20          colorspace_2.1-0           
 [99] BiocGenerics_0.46.0         ggrastr_1.0.1              
[101] tidyselect_1.2.0            processx_3.8.1             
[103] emmeans_1.8.5               bit_4.0.5                  
[105] compiler_4.3.0              git2r_0.32.0               
[107] hdf5r_1.3.8                 DelayedArray_0.25.0        
[109] plotly_4.10.1               bayestestR_0.13.1          
[111] scales_1.2.1                lmtest_0.9-40              
[113] callr_3.7.3                 digest_0.6.31              
[115] goftest_1.2-3               spatstat.utils_3.0-2       
[117] rmarkdown_2.21              XVector_0.40.0             
[119] htmltools_0.5.5             pkgconfig_2.0.3            
[121] MatrixGenerics_1.12.0       highr_0.10                 
[123] fastmap_1.1.1               rlang_1.1.0                
[125] GlobalOptions_0.1.2         htmlwidgets_1.6.2          
[127] shiny_1.7.4                 farver_2.1.1               
[129] jquerylib_0.1.4             zoo_1.8-12                 
[131] jsonlite_1.8.4              mclust_6.0.0               
[133] R.oo_1.25.0                 RCurl_1.98-1.12            
[135] GenomeInfoDbData_1.2.10     parameters_0.21.0          
[137] munsell_0.5.0               Rcpp_1.0.10                
[139] reticulate_1.28-9000        stringi_1.7.12             
[141] zlibbioc_1.46.0             MASS_7.3-59                
[143] plyr_1.8.8                  parallel_4.3.0             
[145] listenv_0.9.0               ggrepel_0.9.3              
[147] deldir_1.0-6                splines_4.3.0              
[149] tensor_1.5                  hms_1.1.3                  
[151] circlize_0.4.15             ps_1.7.5                   
[153] igraph_1.4.2                spatstat.geom_3.1-0        
[155] effectsize_0.8.3            reshape2_1.4.4             
[157] stats4_4.3.0                evaluate_0.20              
[159] BiocManager_1.30.20         ggprism_1.0.4              
[161] tzdb_0.3.0                  httpuv_1.6.9               
[163] MatrixModels_0.5-1          BayesFactor_0.9.12-4.4     
[165] RANN_2.6.1                  polyclip_1.10-4            
[167] scattermore_0.8             rsvd_1.0.5                 
[169] janitor_2.2.0.9000          xtable_1.8-4               
[171] later_1.3.0                 ragg_1.2.5                 
[173] viridisLite_0.4.1           beeswarm_0.4.0             
[175] IRanges_2.34.0              cluster_2.1.4              
[177] timechange_0.2.0            globals_0.16.2             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
